buildtools/gocmd/go.go

```diff
@@ -75,18 +75,25 @@ func (g *Go) List(pkgs, flags []string, vendor bool) ([]Package, error) {
 	if vendor {
 		argv = append([]string{"-mod=vendor"}, argv...)
 	}
-	stdout, stderr, err := exec.Run(exec.Cmd{
+	cmd := exec.Cmd{
 		Name: g.Cmd,
 		Argv: append([]string{"list", "-json"}, argv...),
 		Dir:  g.Dir,
-	})
+	}
+	stdout, stderr, err := exec.Run(cmd)
 	if err != nil && stdout == "" {
 		if strings.Contains(stderr, "build constraints exclude all Go files") {
 			// TODO: add better documentation around this error, and rename it to be
 			// more useful.
 			return nil, errors.New("bad OS/architecture target")
 		}
-		return nil, errors.Errorf("could not run go list: %s (%s)", strings.TrimSpace(stderr), err)
+		return nil, errors.Errorf(
```

Comment:
```suggestion
		return nil, &errors.Error{
			Cause: err,
			Type:  errors.Exec,
			Troubleshooting: fmt.Sprintf("Fossa could not run command `%v` in directory `%s`. Ensure that golang is installed and try running this command on your own.\nstdout: %s\nstderr: %s",
				append([]string{cmd.Name}, cmd.Argv...), cmd.Dir, stdout, strings.TrimSpace(stderr)),
			Link: "https://github.com/fossas/fossa-cli/blob/master/docs/integrations/golang.md#analysis",
		}
```
Feel free to edit this message as you see fit. We just introduced an improved errors package so that we can avoid this exact situation in the future and would like to use it wherever we can.

Reply:
Ah great! I just pushed this change. I pretty much used your exact suggestion with one tweak: I left `stdout` out of the message since this error is only triggered when `stdout` is empty.

Reply:
Good call, thanks!